modules/swagger-codegen/src/main/resources/TypeScript-Angular/api.mustache

```diff
@@ -21,17 +22,24 @@ namespace {{package}} {
                 this.basePath = basePath;
             }
         }
+
+        private extendObj<T1,T2>(objA: T1, objB: T2) {
+            for(let key in objB){
+                if(objB.hasOwnProperty(key)){
+                    objA[key] = objB[key];
+                }
+            }
+            return <T1&T2>objA;
+        }
+
 {{#operation}}
 
         public {{nickname}} ({{#allParams}}{{paramName}}{{^required}}?{{/required}}: {{{dataType}}}, {{/allParams}}extraHttpRequestParams?: any ) : ng.IHttpPromise<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}{}{{/returnType}}> {
-            let path = this.basePath + '{{path}}';
-
-{{#pathParams}}
-            path = path.replace('{' + '{{baseName}}' + '}', String({{paramName}}));
+            const path = this.basePath + '{{path}}'{{#pathParams}}
```

Comment:
I have done this because its more functional style not to modify the path string. Forgive me this little refactoring :)